lib/plugins/wallet/electrum/electrum.js

```diff
@@ -0,0 +1,504 @@
+const ElectrumClient = require('@photon-sdk/rn-electrum-client')
+const NodeCache = require('node-cache')
+const _ = require('lodash/fp')
+const b58 = require('bs58check')
+const bip39 = require('bip39')
+const bitcoin = require('bitcoinjs-lib')
+const coinselect = require('coinselect')
+const ecc = require('tiny-secp256k1')
+const ECPair = require('ecpair').default(ecc)
+const bip32 = require('bip32').default(ecc)
+const mem = require('mem')
+const { default: Queue } = require('p-queue')
+const { unitScale } = require('@lamassu/coins').utils.getCryptoCurrency('BTC')
+
+const BN = require('../../../bn')
+const T = require('../../../time')
+const helpers = require('../helpers')
+const { PromiseObject } = require('../helpers')
+const { loadAccounts } = require('../../../new-settings-loader')
+
+
+const electrumConsts = loadAccounts().then(_.get('electrum'))
+
+const electrumPromise = electrumConsts
+  .then(({ host, port, protocol }) => {
+    const electrum = new ElectrumClient(port, host, protocol)
+    electrum.initElectrum({ client: "someclient x.y.z", version: "1.4" })
+    return electrum
+  })
+
+const networkPromise = electrumConsts
+  .then(({ network }) => ({
+    main: bitcoin.networks['bitcoin'],
+    test: bitcoin.networks['testnet'],
+    regtest: bitcoin.networks['regtest'],
+  })[network])
+
+/* Close the client connection when the server is about to end. */
+_.each(ev => process.once(ev, () => {
+  electrumPromise
+    .then(electrum => electrum.close())
+    .finally(() => process.exit(0))
+}), ['SIGINT', 'SIGTERM', 'exit'])
+
+/*
+ * BIP-125 (§Summary, "Explicit signaling") says any sequence number less than
+ * 0xffffffff-1 explicitly signals RBF. There's also PR#6871 of Bitcoin Core[0]
+ * (linked to from BIP-125).
+ *
+ * [0]: https://github.com/bitcoin/bitcoin/blob/0e935865b9ee3a79fc63f5766074b6f539a0cf85/src/main.cpp#L845-L865
+ *
+ * This is the maximum valid sequence number with RBF on:
+ * `seq <= MAX_RBF_SEQ_NUMBER` means RBF is on, and
+ * `seq > MAX_RBF_SEQ_NUMBER` means RBF is off
+ */
+const MAX_RBF_SEQ_NUMBER = 0xffffffff-2
+
+/* 0x00000000 is the fingerprint of the master key (BIP-32, §Serialization format). */
+const MASTER_FINGERPRINT = Buffer.from([0x00, 0x00, 0x00, 0x00])
+
+const mnemonic = helpers.readMnemonic()
+const wallet_hash = helpers.computeWalletHash(mnemonic)
+
+const GAP_LIMIT = 20
+
+
+const freeAddressIndeces = () =>
+  helpers.freeAddressIndeces('BTC', wallet_hash, ['internal', 'external'])
+
+/* Retrieves the next {in,ex}ternal free index */
+const _nextFreeAddressIndex = (type) => () => freeAddressIndeces().then(_.get(type))
```

Comment:
Am I missing something here?

```suggestion
const _nextFreeAddressIndex = (type) => freeAddressIndeces().then(_.get(type))
```

Reply:
Yeah, the two lines beneath:

https://github.com/lamassu/lamassu-server/blob/c3a938385d499f05346c8b8e2b92a97a6964f5a5/lib/plugins/wallet/electrum/electrum.js#L72-L73

It's just a "delayed" call. I can easily change it if you think it's not very readable.